Error while checking build/O2/fullCI for 723c91ebdc323f249de342e8f44c6d2ccfdf4942 at 2024-06-13 08:59:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/13216-slc8_x86-64/0/Framework/Core/src/Plugin.cxx:70:42: error: 'TFile' was not declared in this scope
/sw/SOURCES/O2/13216-slc8_x86-64/0/Framework/Core/src/Plugin.cxx:70:49: error: 'f' was not declared in this scope
/sw/SOURCES/O2/13216-slc8_x86-64/0/Framework/Core/src/Plugin.cxx:71:1: error: expected ',' or ';' before '{' token
/sw/SOURCES/O2/13216-slc8_x86-64/0/Framework/Core/src/Plugin.cxx:109:93: error: no match for call to '(std::vector<std::__cxx11::basic_string<char> >) (std::string&)'
/sw/SOURCES/O2/13216-slc8_x86-64/0/Framework/Core/src/Plugin.cxx:109:130: error: no matching function for call to 'o2::framework::ConfigParamSpec::ConfigParamSpec(<brace-enclosed initializer list>)'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13216/723c91ebdc323f249de342e8f44c6d2ccfdf4942/build_O2_fullCI/pretty.html).

--- a/Framework/Core/src/Plugin.cxx
+++ b/Framework/Core/src/Plugin.cxx
@@ -14,6 +14,7 @@
 #include "Framework/Logger.h"
 #include "Framework/Capability.h"
 #include "Framework/Signpost.h"
+#include "Framework/VariantJSONHelpers.h"
 #include <string_view>
 
 O2_DECLARE_DYNAMIC_LOG(capabilities);
@@ -86,7 +87,13 @@ struct DiscoverMetadataInCommandLine : o2::framework::ConfigDiscoveryPlugin {
           std::string value = argv[i + 1];
           O2_SIGNPOST_EVENT_EMIT(capabilities, sid, "DiscoverMetadataInCommandLine",
                                  "Found %{public}s with value %{public}s.", key.c_str(), value.c_str());
-          results.push_back(ConfigParamSpec{key, VariantType::String, value, {"Metadata in command line"}});
+          if (key == "aod-metadata-tables") {
+            std::stringstream is(value);
+            auto arrayValue = VariantJSONHelpers::read<VariantType::ArrayString>(is);
+            results.push_back(ConfigParamSpec{key, VariantType::ArrayString, arrayValue, {"Metadata in command line"}});
+          } else {
+            results.push_back(ConfigParamSpec{key, VariantType::String, value, {"Metadata in command line"}});
+          }
         }
         return results;
       }};

